src/components/AuthContainer.js

```diff
@@ -19,7 +19,6 @@ const AuthContainer = ({ children }) => {
   const seenAzurePreview = useStore(azurePreviewStore) || false;
   const authspinner = () => h(centeredSpinner, { style: { position: 'fixed' } });
   return Utils.cond(
-    [isSignedIn === undefined && !isPublic, authspinner],
```

Comment:
Confirmed that this is happening. I can see it by adding latency to requests in dev tools. Or it can also be observed by adding a console log to the Sign In page.

Reply:
Added a new property `isSigningIn` which is set to true when clicking signIn and set to false once the `oidc-client` responds with a user. I added this check back in using `isSigningIn` rather than `isSignedIn === undefined`

Reply:
`isSigningIn` here doesn't address the issue. It handles a different case: while the request triggered when the user clicks the sign in button is ongoing. I'm not sure we need to keep track of that _in global state_.

When Terra UI is first loaded, we don't know whether the user is signed in or not until an API request returns. By initializing `isSignedIn` to false, we end up showing the Sign In page while that request is going on. For keeping track of that, the tri-state approach (true, false, or undefined) of `isSignedIn` has a nice feature: it makes invalid states unrepresentable.